docs/reference/master-configuration-reference.md

```diff
@@ -0,0 +1,111 @@
+---
+title: "Master config reference"
+layout: default
+sort: 4
+---
+
+# Configuration file reference of nfd-master
+{: .no_toc}
+
+## Table of contents
+{: .no_toc .text-delta}
+
+1. TOC
+{:toc}
+
+---
+
+See the
+[sample configuration file](https://github.com/kubernetes-sigs/node-feature-discovery/blob/{{site.release}}/deployment/components/master-config/nfd-master.conf.example)
+for a full example configuration.
+
+## noPublish
+
+`noPublish` option disables updates to the Node objects in the Kubernetes
+API server, making a "dry-run" flag for nfd-master. No Labels, Annotations or
+ExtendedResources of nodes are updated.
+
+Default: `false`
+
+Example:
+
+```yaml
+noPublish: true
+```
+
+## extraLabelNs
+`extraLabelNs` specifies a comma-separated list of allowed feature
+label namespaces. This option can be used to allow
+other vendor or application specific namespaces for custom labels from the
+local and custom feature sources, even though these labels were denied using
+the `denyLabelNs` parameter.
+
+The same namespace control and this option applies to Extended Resources (created
+with `resourceLabels`), too.
+
+Default: *empty*
+
+Example:
+
+```yaml
+extraLabelNs: {"added.ns.io","added.kubernets.io"}
+```
+
+## denyLabelNs
+`denyLabelNs` specifies an array list of excluded
+label namespaces. By default, nfd-master allows creating labels in all
+namespaces, excluding `kubernetes.io` namespace and its sub-namespaces
+(i.e. `*.kubernetes.io`). However, you should note that
+`kubernetes.io` and its sub-namespaces are always denied.
+This option can be used to exclude some vendors or application specific
+namespaces.
+Note that the namespaces `feature.node.kubernetes.io` and `profile.node.kubernetes.io`
+and their sub-namespaces are always allowed and cannot be denied.
+
+Default: *empty*
+
+Example:
+
+```yaml
+denyLabelNs: {"denied.ns.io","denied.kubernetes.io"}
```

Comment:
just a note that we need to fix all of these when #1087 gets merged and we rebase this
```suggestion
denyLabelNs: ["denied.ns.io","denied.kubernetes.io"]
```